src/check/canonicalize.zig

```diff
@@ -419,21 +483,24 @@ pub fn canonicalize_expr(
         .ident => |e| {
             const region = self.tokenizedRegionToRegion(e.region);
             if (self.parse_ir.tokens.resolveIdentifier(e.token)) |ident| {
-                if (self.scope.levels.lookup(&self.can_ir.env.idents, .ident, ident)) |pattern_idx| {
-                    // We found the ident in scope, lookup to reference the pattern
-                    const expr_idx =
-                        self.can_ir.store.addExpr(CIR.Expr{ .lookup = .{
-                            .pattern_idx = pattern_idx,
+                switch (self.scopes.lookup(&self.can_ir.env.idents, .ident, ident)) {
+                    .found => |pattern_idx| {
+                        // We found the ident in scope, lookup to reference the pattern
+                        const expr_idx =
+                            self.can_ir.store.addExpr(CIR.Expr{ .lookup = .{
+                                .pattern_idx = pattern_idx,
+                                .region = region,
+                            } });
+                        _ = self.can_ir.setTypeVarAtExpr(expr_idx, Content{ .flex_var = null });
```

Comment:
```suggestion
```

I don't think this should be necessary!

In type Hindley-Milner type inference, if I have `a = 5` and then later I reference `a + 1`, the type of that `a` expression in `a + 1` (so, the `lookup` on `a`) is going to be equal to the type of the `a` in the pattern.

Since CIR index and type var are the same number in this design, we shouldn't need to do anything else. If we come across this node and want to know what its type is, `pattern_idx` immediately tells us the answer - that's its type, because their types are equal, because it's a lookup! 😄 

cc @jaredramirez

Reply:
If you don't mind @jaredramirez I'll leave this for you to review/update in another PR as I don't fully understand it.